src/tests/ftest/util/AgentUtils.py

```diff
@@ -78,18 +87,33 @@ def run_agent(basepath, client_list, server_list):
     basepath    --root directory for DAOS repo or installation
     client_list --those nodes that are acting as compute nodes in the test
     server_list --those nodes that are acting as server nodes in the test
-
     """
-    if not node_setup_okay(client_list, NodeListType.CLIENT):
-        raise AgentFailed("A compute node isn't configured properly")
+    sessions = {}
+
+    user = getpass.getuser()
+
+    retcode, node, agent_dir = node_setup_okay(server_list, NodeListType.SERVER)
+    if not retcode:
+        raise AgentFailed("Server node " + node + " does not have directory "
+                          + agent_dir + " set up correctly for user "
+                          + user + ".")
+
+    # if empty client list, 'self' is effectively client
+    if client_list is None:
+        client_list = [socket.gethostname()]
```

Comment:
Is this going to be the _hostname_ + _domainname_ and is it going to agree with the client list passed to `stop_agent()`?

https://github.com/daos-stack/daos/blob/34f9bb1525a99dc560136c03c21b186444e21c14/src/tests/ftest/util/AgentUtils.py#L160

I suspect you are putting _hostname.domainname_ into `sessions` (by way of `client_list`) and trying to find the session later with just _hostname_.

You probably want to strip any _.domainname_ off of what you get from `socket.gethostname()`.

Reply:
Changed this up, the behavior now is to strip out everything after a '.' if one shows up, and if no '.' shows up, the full string is returned. This scans with what I'm seeing suggested online, if there's a better way, please feel free to suggest here.